cmake/onnxruntime_python.cmake

```diff
@@ -96,11 +70,25 @@ target_link_libraries(onnxruntime_pybind11_state PRIVATE
     ${PROVIDERS_ACL}
     ${PROVIDERS_ARMNN}
     ${PROVIDERS_ROCM}
-    onnxruntime_optimizer
     onnxruntime_providers
     onnxruntime_util
     ${onnxruntime_tvm_libs}
     onnxruntime_framework
+)
+
+if (onnxruntime_ENABLE_TRAINING)
+  list(APPEND onnxruntime_pybind11_state_link_targets onnxruntime_training)
```

Comment:
Apparently onnxruntime_training depends on onnxruntime_framework. Why do you put it after that?

Reply:
will fix it.

Reply:
I observed this, just did not want to introduce another if branch. updated.

Reply:
I still don't understand, why do you append onnxruntime_training to the end?

Reply:
Does onnxruntime_training use any function from onnxruntime_util or any EPs?

Reply:
EPs depends on onnxruntime_training.